main.go

```diff
@@ -147,20 +161,29 @@ func cleanWindows(p string) string {
 func main() {
 
 	var headers headerList
+	var proxyURL url.URL
 
 	outDir := flag.String("output", "", "Source file output directory - REQUIRED")
-	url := flag.String("url", "", "URL or path to the Sourcemap file - REQUIRED")
+	urlflag := flag.String("url", "", "URL or path to the Sourcemap file - REQUIRED")
+	proxy := flag.String("proxy", "", "Explicity HTTP client proxy")
 	help := flag.Bool("help", false, "Show help")
 	insecure := flag.Bool("insecure", false, "Ignore invalid TLS certificates")
 	flag.Var(&headers, "header", "A header to send with the request, similar to curl's -H. Can be set multiple times, EG: \"./sourcemapper --header \"Cookie: session=bar\" --header \"Authorization: blerp\"")
 	flag.Parse()
 
-	if *help || *url == "" || *outDir == "" {
+	if *help || *urlflag == "" || *outDir == "" {
 		flag.Usage()
 		os.Exit(1)
 	}
+	if *proxy != "" {
+		p, err := url.Parse(*proxy)
+		if err != nil {
+			log.Println(err)
```

Comment:
this should `log.Fatalln()` instead. This code will continue to set `p` even if there is a parse issue. Also, `isURL` is defined and already used in this file, why not use that here too?

Reply:
The URL handling one was actually deliberate as the `isURL` is for checking for HTTP requests, and the Go `ProxyURL` handler can use `socks5://`. I figured it didn't make any sense to mix up the two, but if you think it'd be make more sense to have that in there I'd b happy to switch it up.

Reply:
Makes sense to me :D